src/Http/Routing/src/Constraints/FileNameRouteConstraint.cs

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Routing.Constraints
+{
+    /// <summary>
+    /// Constrains a route parameter to represent only filename values.
```

Comment:
IMO it worth commenting that this route constraint only checks whether the parameter is formatted like a file name, and doesn't check whether the file exists on disk. That was my first thought when I saw the class name.

Reply:
Sure, I agree. I mentioned it for `MapFallback` docs, but not here.